HTTP_server.py

```diff
@@ -0,0 +1,128 @@
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+
+            recieve_total = ""
+            buffersize = 32
+            finished = 0
+            while not finished:
+                recieve = conn.recv(buffersize)
+                if len(recieve) < buffersize:
+                    recieve_total += recieve
+                    finished = 1
+                else:
+                    recieve_total += recieve
+
+            if recieve_total:
+                try:
+                    uri = parse_request(recieve_total)
+                    info = resolve_uri(uri)
+                    response = response_ok(info)
+                except RequestError, msg:
+                    errors = str(msg).strip("'").split(' ', 1)
+                    response = response_error(errors[0], errors[1])
+                conn.sendall(response)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def response_ok(*args):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: {}'.format(args[0][0])
+    body = '''<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n</head>\n<body>\n{}</body>\n</html>'''.format(args[0][1])
+    response_list = [first_line, timestamp, content_header, ' ', body, '\r\n']
+    return '\r\n'.join(response_list)
+
+
+def response_error(error_code, error_message):
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/plain'
+    body = '{} {}\n'.format(error_code, error_message)
+    body = '''<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n</head>\n<body>\n{}</body>\n</html>'''.format(body)
+    response_list = [first_line, timestamp, content_header, ' ', body, '\r\n']
+    return '\r\n'.join(response_list)
+
+
+def parse_request(client_request):
+    mup_line = client_request.splitlines()[0]
+    mup = mup_line.split(' ')
+    uri = error_mup(mup)
+    return uri
+
+
+def error_mup(mup):
+    http_response_codes = {'405': 'Method Not Allowed',
+                           '505': 'HTTP Version Not Supported'}
+
+    if mup[0] != 'GET':
+        error_key = '405'
+        msg = "{} {}".format(error_key, http_response_codes[error_key])
+        raise RequestError(msg)
+    elif mup[2] != 'HTTP/1.1':
+        error_key = '505'
+        msg = "{} {}".format(error_key, http_response_codes[error_key])
+        raise RequestError(msg)
+    else:
+        return mup[1]
+
+
+def resolve_uri(uri):
+    if os.path.isdir(uri):
+        body = gen_list(uri)
+        content_type = 'text/html'
+        info = (content_type, body)
+    elif os.path.isfile(uri):
+        body = gen_text(uri)
+        content_type = 'text/html'
```

Comment:
This is not right.  Not every item you view will be an html page.  figure out how to get the right content-type value